src/test/java/com/wojteknier03/clinicmedical/service/AppointmentServiceTest.java

```diff
@@ -128,6 +128,58 @@ private Appointment createAppointment(Long id) {
         return appointment;
     }
 
+    @Test
+    void assignPatientToAppointment_ValidIds_ExceptionThrown(){
+        //given
+        Patient patient = new Patient();
+        Long appointmentId = 1L;
+        Long patientId = 2L;
+        Appointment appointment = createAppointment(appointmentId);
+
+        when(appointmentRepository.findById(appointmentId)).thenReturn(Optional.of(appointment));
+        when(patientRepository.findById(patientId)).thenReturn(Optional.of(patient));
+
+        //when
+        appointmentService.assignPatientToAppointment(appointmentId, patientId);
+
+        //then
+        verify(appointmentRepository, times(1)).save(appointment);
+        verify(appointmentRepository, times(1)).findById(appointmentId);
+        verify(patientRepository, times(1)).findById(patientId);
```

Comment:
Odnosząc się do tego co napisałem powyżej - testy powinno się pisać w taki sposób żeby sprawdzały **zachowania** a nie **implementację**. Assercje w rodzaju ile i jakie wywołania metod są w serwisie, betonują implementację i przez to trudniej wprowadzać zmiany w kodzie. 
Ja bym ten test napisał tak (pseudokod):
```
void shouldAssignPatientToAppointmentIfPatienAndAppointmentExists(){

// given
var patient = //create patient
patientRepository.save(patient);
var appointment = //create appointment
appointmentRepository.save(patient);

// when
appointmentService.assignPatientToAppointment(appointmentId, patientId);

// then
patientRepository.findById(patientId).getAppointments() .... sprawdzenie czy jest tam appointmentId
appointmentRepository.findById(appointmentId).getPatient() .... sprawdzenie pacjent jest ten sam
}
```